src/dynprog.rs

```diff
@@ -196,6 +261,73 @@ mod tests {
         );
     }
 
+    #[test]
+    fn edit_script_tests() {
+        let applet: Vec<char> = "applet".chars().collect();
+        let pineapple: Vec<char> = "pineapple".chars().collect();
+        let aplpet: Vec<char> = "aplpet".chars().collect();
+
+        let levenshtein = Levenshtein::new(1, 1, 1);
+        assert_eq!(
+            hashset![
+                vec![
+                    IndexedOperation::new(Delete(1), 0, 0),
+                    IndexedOperation::new(Delete(1), 1, 0),
+                    IndexedOperation::new(Delete(1), 2, 0),
+                    IndexedOperation::new(Delete(1), 3, 0),
+                    IndexedOperation::new(Match, 4, 0),
+                    IndexedOperation::new(Match, 5, 1),
+                    IndexedOperation::new(Match, 6, 2),
+                    IndexedOperation::new(Match, 7, 3),
+                    IndexedOperation::new(Match, 8, 4),
+                    IndexedOperation::new(Insert(1), 9, 5),
+                ],
+            ],
+            levenshtein.align(&pineapple, &applet).edit_scripts()
+        );
+
+        assert_eq!(
+            hashset![
+                vec![
+                    IndexedOperation::new(Match, 0, 0),
+                    IndexedOperation::new(Match, 1, 1),
+                    IndexedOperation::new(Substitute(1), 2, 2),
+                    IndexedOperation::new(Substitute(1), 3, 3),
+                    IndexedOperation::new(Match, 4, 4),
+                    IndexedOperation::new(Match, 5, 5),
+                ],
+                vec![
+                    IndexedOperation::new(Match, 0, 0),
+                    IndexedOperation::new(Match, 1, 1),
+                    IndexedOperation::new(Delete(1), 2, 2),
+                    IndexedOperation::new(Match, 3, 2),
+                    IndexedOperation::new(Insert(1), 4, 3),
+                    IndexedOperation::new(Match, 4, 4),
+                    IndexedOperation::new(Match, 5, 5),
+                ],
+                vec![
+                    IndexedOperation::new(Match, 0, 0),
+                    IndexedOperation::new(Delete(1), 1, 1),
+                    IndexedOperation::new(Match, 2, 1),
+                    IndexedOperation::new(Match, 3, 2),
+                    IndexedOperation::new(Insert(1), 4, 3),
+                    IndexedOperation::new(Match, 4, 4),
+                    IndexedOperation::new(Match, 5, 5),
+                ],
+                vec![
+                    IndexedOperation::new(Match, 0, 0),
+                    IndexedOperation::new(Match, 1, 1),
+                    IndexedOperation::new(Insert(1), 2, 2),
+                    IndexedOperation::new(Match, 2, 3),
+                    IndexedOperation::new(Delete(1), 3, 4),
+                    IndexedOperation::new(Match, 4, 4),
+                    IndexedOperation::new(Match, 5, 5),
+                ],
+            ],
+            levenshtein.align(&applet, &aplpet).edit_scripts()
+        );
```

Comment:
Add test for comparing identical strings like "applet" - "applet"?

Reply:
To ensure that it doesn't accidentally produce > 1 alignment? Good idea!